addons/web/static/src/views/fields/many2one/many2one_field.js

```diff
@@ -179,7 +179,15 @@ export class Many2OneField extends Component {
         return this.props.canOpen && !!this.value && !this.state.isFloating;
     }
     get context() {
-        return this.props.context;
+        const { name, context, record } = this.props;
+        if (
+            context.active_model &&
+            context.active_model !== record.fields[name].context.active_model
+        ) {
+            context.active_model = record.fields[name].context.active_model;
+        }
+
+        return context;
```

Comment:
this doens't make sense. You basically remove the `active_model` key all the time, but it is a relevant information. And having another `active_model` in the field context doesn't make sense either, as the active_id would still reference a record of the initial model. I don't know what the issue is, but I assume it comes from the business logic of that specific flow

Reply:
or all `active_model`, `active_id` and `active_ids` shouldn't be transferred. I'll investigate a bit more.